### PR TITLE
Move FSImpl::CloseFile() to BfsFileImpl::Close()

### PR DESCRIPTION
**Pull request by yvxiang**
或者直接去掉FSImpl::CloseFile()?

### CODE REVIEW

File: src/sdk/bfs.cc [hunk 1]
```diff
@@ -998,7 +970,23 @@ bool BfsFileImpl::Close() {
     _chunkserver = NULL;
     LOG(DEBUG, "File %s closed", _name.c_str());
     _closed = true;
-    return !_bg_error;
+    if (need_report_finish) {
```
Comment (bluebore) [inline src/sdk/bfs.cc]: 如果_bg_error为true，也调用FinishBlock？

**Comment by bluebore** [comment]
102

**Comment by bluebore** [comment]
更新后，日志中出现大量的fail
01/05 07:24:10.081034 13925 [src/sdk/bfs.cc:987] Close file fail
01/05 07:24:20.663830 16494 [src/sdk/bfs.cc:987] Close file fail
01/05 07:24:30.693560 16648 [src/sdk/bfs.cc:987] Close file fail
01/05 07:24:39.925829 16673 [src/sdk/bfs.cc:987] Close file fail
01/05 07:24:59.897535 18858 [src/sdk/bfs.cc:987] Close file fail
01/05 07:25:20.372789 22305 [src/sdk/bfs.cc:987] Close file fail
01/05 07:25:40.420457 22562 [src/sdk/bfs.cc:987] Close file fail
01/05 07:25:50.193686 22694 [src/sdk/bfs.cc:987] Close file fail
01/05 07:26:00.476429 25238 [src/sdk/bfs.cc:987] Close file fail
01/05 07:26:10.473310 25366 [src/sdk/bfs.cc:987] Close file fail
01/05 07:26:19.717606 25367 [src/sdk/bfs.cc:987] Close file fail
01/05 07:26:39.812202 27929 [src/sdk/bfs.cc:987] Close file fail
01/05 07:27:00.339768 30828 [src/sdk/bfs.cc:987] Close file fail
01/05 07:27:10.181595 415 [src/sdk/bfs.cc:987] Close file fail

**Comment by yvxiang** [comment]
#117  
看看打出来什么

**Pull request merged**